routers/repo/issue.go

```diff
@@ -1447,6 +1448,11 @@ func ChangeIssueReaction(ctx *context.Context, form auth.ReactionForm) {
 
 	switch ctx.Params(":action") {
 	case "react":
+		if !util.IsStringInSlice(form.Content, setting.UI.Reactions) {
+			log.Error("ChangeIssueReaction: Reaction '%s' is no allowed reaction", form.Content)
```

Comment:
This should return an error the user, not just log it.  This func returns a JSON response that can contain any error you need to convey.

Also, small wording nit.
```suggestion
			log.Error("ChangeIssueReaction: '%s' is not an allowed reaction", form.Content)
```

Reply:
@jolheiser same as i told @zeripath if you fall into this error - a custom template is wrong ... -> admin responce not the user one

wording nit's -> always wellcome (I'm bad at it)

Reply:
That may be so, but currently it will just fail silently and the user won't know why. That leads to frustrating user experience.

Reply:
@jolheiser if you show me how to do -> flash + redirect is a worse user experience  (like to react and get redirected to a error page :D )

Reply:
No, there should be no redirecting. Instead, an error could be returned to the page in JSON form that the page could then use to alert the user.  
If you look at the final return of this func, it's a JSON response with the HTML as a field. Rather, this could return an error field that the template JS could check to see if there is an error or not.

Reply:
@jolheiser does 0e406a7cece355bcf7b31e9b458b1e8fb72b6581 resolve your issue ?

Reply:
It's not what I had in mind, but I see we are already using a ServerError in that function so it's no worse than that.  
To clarify what I am meaning, the return of those functions is 
```go
ctx.JSON(200, map[string]interface{}{
		"html": html,
	})
```
And the JS uses the `html` to populate the reaction bar.  
  
What this _could_ do is return errors via
```go
ctx.JSON(500, map[string]interface{}{
		"html": html,
                "error": "some error message",
	})
```
and the JS could then give a nice error message to the user instead.